app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -623,8 +624,24 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 refreshCurrentView(false);
                 break;
         }
+        mFormController.setPendingCalloutFormIndex(null);
     }
-    
+
+    private IntentWidget tryToGetIntentWidget() {
+        IntentWidget bestMatch = null;
+
+        //Ugh, copied from the odkview mostly, that's stupid
+        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
+            //Figure out if we have a pending intent widget
+            if (q instanceof IntentWidget) {
+                if(((IntentWidget) q).isWaitingForBinaryData() || bestMatch == null) {
```

Comment:
Hey,

I think this whole function can be replaced with the calls below? Looks like you're still relying on the old system here before jumping into the new logic